stdlib/websub/src/main/java/org/ballerinalang/net/websub/nativeimpl/InitWebSubSubscriberServiceEndpoint.java

```diff
@@ -110,14 +120,93 @@ public void execute(Context context) {
+                case TOPIC_ID_PAYLOAD_KEY:
+                    populateResourceDetailsByPayload(payloadKeyResourceMap, resourceDetails);
+                    break;
+                default:
+                    populateResourceDetailsByHeaderAndPayload(headerAndPayloadKeyResourceMap, resourceDetails);
+                    if (headerResourceMap != null) {
+                        populateResourceDetailsByHeader(headerResourceMap, resourceDetails);
+                    }
+                    if (payloadKeyResourceMap != null) {
+                        populateResourceDetailsByPayload(payloadKeyResourceMap, resourceDetails);
+                    }
+                    break;
+            }
+        }
+        return resourceDetails;
+    }
+
+    private static void populateResourceDetailsByHeader(BMap<String, BValue> headerResourceMap,
+                                                        HashMap<String, String[]> resourceDetails) {
+        headerResourceMap.getMap().values().forEach(value -> {
+            BRefValueArray resourceDetailTuple = (BRefValueArray) value;
+            String resourceName = resourceDetailTuple.getBValue(0).stringValue();
+            BStructureType paramDetails =
+                    (BStructureType) ((BTypeDescValue) (resourceDetailTuple).getBValue(1)).value();
+            resourceDetails.put(resourceName,
+                                new String[]{paramDetails.getPackagePath(), paramDetails.getName()});
+        });
+    }
+
+    private static void populateResourceDetailsByPayload(BMap<String, BMap<String, BValue>> payloadKeyResourceMap,
+                                                         HashMap<String, String[]> resourceDetails) {
+        payloadKeyResourceMap.getMap().values().forEach(mapByKey -> {
+            mapByKey.getMap().values().forEach(value -> {
+                BRefValueArray resourceDetailTuple = (BRefValueArray) value;
+                String resourceName = resourceDetailTuple.getBValue(0).stringValue();
+                BStructureType paramDetails =
+                        (BStructureType) ((BTypeDescValue) (resourceDetailTuple).getBValue(1)).value();
+                resourceDetails.put(resourceName,
+                                    new String[]{paramDetails.getPackagePath(), paramDetails.getName()});
+            });
+        });
+    }
+
+    private static void populateResourceDetailsByHeaderAndPayload(BMap<String, BMap<String, BMap<String, BValue>>>
+                                                                          headerAndPayloadKeyResourceMap,
+                                                                  HashMap<String, String[]> resourceDetails) {
+        headerAndPayloadKeyResourceMap.getMap().values().forEach(mapByHeader -> {
+            mapByHeader.getMap().values().forEach(mapByKey -> {
+                mapByKey.getMap().values().forEach(value -> {
+                    BRefValueArray resourceDetailTuple = (BRefValueArray) value;
```

Comment:
Shall we move this redundant code in to another method as follows?

```ballerina
private static void populateResourceDetails(HashMap<String, String[]> resourceDetails, BRefValueArray resourceDetailTuple) {
        String resourceName = resourceDetailTuple.getBValue(0).stringValue();
        BStructureType paramDetails =
                (BStructureType) ((BTypeDescValue) (resourceDetailTuple).getBValue(1)).value();
        resourceDetails.put(resourceName,
                new String[]{paramDetails.getPackagePath(), paramDetails.getName()});
    }
```

Reply:
Fixed with e631a56, thanks.